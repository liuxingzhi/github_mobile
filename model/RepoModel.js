const emptyRepoData = {
    name: '',
    url: '',
    description: '',
    updatedAt: '',
    primaryLanguage: {
        name: '',
        color: '#FFFFFF', // White, invisible by default
    },
    stargazers: {
        totalCount: 0,
    },
    forkCount: 0
}

class RepoModel {
    constructor(repoData = emptyRepoData) {
        if (repoData == null){
            repoData = emptyRepoData
        }
        this.name = repoData.name != null ? repoData.name.trim() : ''
        this.url = repoData.url != null ? repoData.url : ''
        this.description = repoData.description != null ? repoData.description : ''
        this.updatedAt = repoData.updatedAt != null ? repoData.updatedAt : ''
        this.primaryLanguage = repoData.primaryLanguage != null ? repoData.primaryLanguage.name : ''
        this.primaryLanguageColor = repoData.primaryLanguage != null ? repoData.primaryLanguage.color : '#ffffff'
        this.starCount = repoData.stargazers != null ? repoData.stargazers.totalCount : 0
        this.forkCount = repoData.forkCount != null ? repoData.forkCount : 0
    }

    searchKeyCaseInsensitive(key){
        return this.name.toLowerCase().includes(key.toLowerCase());
    }
}

export {RepoModel, emptyRepoData}