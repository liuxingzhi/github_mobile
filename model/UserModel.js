const emptyUserData = {
    avatarUrl: '',
    name: '',
    login: '',
    bio: '',
    websiteUrl: '',
    email: '',
    repositories: {totalCount: ''},
    followers: {totalCount: ''},
    following: {totalCount: ''},
    createDate: '',
}

class UserModel {
    constructor(userData = emptyUserData) {
        if (userData == null){
            userData = emptyUserData;
        }
        this.profileImage = userData.avatarUrl != null ? userData.avatarUrl : ''
        this.name = userData.name != null ? userData.name : ''
        this.username = userData.login != null ? userData.login : ''
        this.bio = userData.bio
        this.website = userData.websiteUrl != null ? userData.websiteUrl : ''
        this.email = userData.email != null ? userData.email : ''
        this.publicReposCount = userData.repositories != null ? userData.repositories.totalCount : 0
        this.followersCount = userData.followers != null ? userData.followers.totalCount : 0
        this.followingCount = userData.following != null ? userData.following.totalCount : 0
        this.createDate = userData.createdAt
    }

    set createDate(createdAt) {
        this._createDate = createdAt != null ? createdAt : ''
    }

    get createDate() {
        return this._createDate.split('T')[0]
    }

    set bio(userBio) {
        this._bio = userBio != null ? userBio : ''
    }

    get bio() {
        return this._bio.trim()
    }

    searchKeyCaseInsensitive(key){
        return this.username.toLowerCase().includes(key.toLowerCase());
    }
}

export {UserModel, emptyUserData}