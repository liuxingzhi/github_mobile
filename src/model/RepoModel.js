/**
 * default empty data structure for repo
 * @type {{stargazers: {totalCount: number}, name: string, description: string, forkCount: number, url: string, primaryLanguage: {color: string, name: string}, updatedAt: string}}
 */
const emptyRepoData = {
    name: '',
    url: '',
    description: '',
    updatedAt: '',
    primaryLanguage: {
        name: '',
        color: '#FFFFFF', // White, invisible by default
    },
    stargazers: {
        totalCount: 0,
    },
    forkCount: 0
}

/**
 * Class model for repository, used by repo screen
 */
class RepoModel {
    constructor(repoData = emptyRepoData) {
        if (repoData == null){
            repoData = emptyRepoData
        }
        this.name = repoData.name != null ? repoData.name.trim() : ''
        this.url = repoData.url != null ? repoData.url : ''
        this.description = repoData.description != null ? repoData.description : ''
        this.updatedAt = repoData.updatedAt != null ? repoData.updatedAt : ''
        this.primaryLanguage = repoData.primaryLanguage != null ? repoData.primaryLanguage.name : ''
        this.primaryLanguageColor = repoData.primaryLanguage != null ? repoData.primaryLanguage.color : '#ffffff'
        this.starCount = repoData.stargazers != null ? repoData.stargazers.totalCount : 0
        this.forkCount = repoData.forkCount != null ? repoData.forkCount : 0
        this.owner = repoData.owner != null ? repoData.owner.login : ''
    }

    searchKeyCaseInsensitive(key){
        return this.name.toLowerCase().includes(key.toLowerCase());
    }
}

export {RepoModel, emptyRepoData}