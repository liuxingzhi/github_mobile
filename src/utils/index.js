// export { loadFollowerData, loadFollowingData, loadRepoData, loadUserData } from "./service.js";
// export { githubQuery, parseNodesToModel } from "./util.js";

export * from "./service.js";
export * from "./util.js";