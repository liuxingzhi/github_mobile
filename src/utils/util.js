import * as ENV from "../env.js"
import fetch from 'cross-fetch'
import {emptyUserData, UserModel} from "../model/UserModel.js"

/**
 * A wrap to the github API call
 * @param query
 * @param token
 * @returns {Promise<Response|{}>}
 */
async function githubQuery(query, token) {
    let url = 'https://api.github.com/graphql'
    return await fetch(url, {
        method: 'POST',
        headers: {
            Authorization: `bearer ${ENV.USER_AUTH_TOKEN}`,
            'Content-Type': 'application/json',
        },
        body: JSON.stringify({
            query: query,
        }),
    }).then(response => {
        if (response.ok) {
            return response.json();
        } else {
            switch (response.status){
                case 401:
                    console.log(new Error(`401 Unauthorized`))
                case 403:
                    console.log (new Error(`403 Forbidden`))
                default:
                    return response.json()
            }
        }
    }).catch(error => {
        console.log(error)
        return {}
    })
}

/**
 * A helper function to convert nodes array to corresponding list of model
 * @param nodesArray
 * @param Model: A class instance, the returned model list will be of the class type passed in
 * @returns {[]}
 */
function parseNodesToModel(nodesArray, Model) {
    let modelList = []
    for (const node of nodesArray) {
        modelList.push(new Model(node))
    }
    return modelList
}

export {githubQuery, parseNodesToModel}
