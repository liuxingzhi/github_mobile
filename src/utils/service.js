/**
 * This file contain 4 different kinds of services,
 * loadUserData
 * loadFollowerData
 * loadFollowingData
 * loadRepoData
 */
import {githubQuery, parseNodesToModel} from "./util.js";
import {emptyUserData, UserModel} from "../model/UserModel.js";
import {emptyRepoData, RepoModel} from "../model/RepoModel.js";


async function loadUserData(username) {
    let query = `query { 
                user(login:"${username}") { 
                    login
                    bio
                    avatarUrl
                    websiteUrl
                    createdAt
                    name
                    email
                    repositories(first: 100) {
                        totalCount
                    }
                    followers {
                        totalCount
                    }
                    following {
                        totalCount
                    }
                }
        }`

    try {
        // load data
        let response = await githubQuery(query);
        let userData = response.data.user
        return new UserModel(userData)
    }catch (error){
        // use empty data when error
        // console.log(error)
        return new UserModel(emptyUserData)
    }
}

async function loadRepoData(username) {
    let query = `query {
                        user(login:"${username}") {
                            repositories(first: 100, orderBy: {field: UPDATED_AT, direction: DESC}){
                                nodes{
                                    owner{
                                      login
                                    }
                                    name
                                    url
                                    description
                                    updatedAt
                                    primaryLanguage{
                                        name
                                        color
                                    }
                                    stargazers{
                                        totalCount
                                    }
                                    forkCount
                                }
                            }
                        }
                    }`
    try {
        // load data
        let response = await githubQuery(query);
        let repoDataArray = response.data.user.repositories.nodes
        let repos = parseNodesToModel(repoDataArray, RepoModel)
        return repos
    }catch (error){
        // use empty data when error
        // console.log(error)
        return [new RepoModel(emptyRepoData)]
    }
}

async function loadFollowerData(username) {
    let query = `
        {
          user(login: "${username}") {
            login
            bio
            followers(first: 100) {
              nodes {
                login
                bio
                name
                avatarUrl
                repositories {
                  totalCount
                }
              }
            }
          }
        }`

    try {
        // load data
        let response = await githubQuery(query);
        let userDataArray = response.data.user.followers.nodes
        let userList = parseNodesToModel(userDataArray, UserModel)
        userList.sort(function (a, b) {
            return a.publicReposCount - b.publicReposCount
        })
        return userList
    }catch (error){
        // use empty data when error
        // console.log(error)
        return [new UserModel(emptyUserData)]
    }
}

async function loadFollowingData(username) {
    let query = `
        {
          user(login: "${username}") {
            login
            bio
            following(first: 100) {
              nodes {
                login
                name
                bio
                avatarUrl
                repositories {
                  totalCount
                }
              }
            }
          }
        }`

    try {
        // load data
        let response = await githubQuery(query);
        let userDataArray = response.data.user.following.nodes
        let userList = parseNodesToModel(userDataArray, UserModel)
        userList.sort(function (a, b) {
            return a.publicReposCount - b.publicReposCount
        })
        return userList
    }catch (error){
        // use empty data when error
        // console.log(error)
        return [new UserModel(emptyUserData)]
    }
}

export {loadFollowerData, loadFollowingData, loadRepoData, loadUserData}