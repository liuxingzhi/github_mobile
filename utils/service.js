/**
 * This file contain 4 different kinds of services,
 * loadUserData
 * loadFollowerData
 * loadFollowingData
 * loadRepoData
 */
import {githubQuery, parseNodesToModel} from "./util";
import {emptyUserData, UserModel} from "../model/UserModel";
import {emptyRepoData, RepoModel} from "../model/RepoModel";

function loadUserData(username) {
    let query = `query { 
                user(login:"${username}") { 
                    login
                    bio
                    avatarUrl
                    websiteUrl
                    createdAt
                    name
                    email
                    repositories(first: 100) {
                        totalCount
                    }
                    followers {
                        totalCount
                    }
                    following {
                        totalCount
                    }
                }
        }`

    githubQuery(query)
        .then(response => {
            let userData = response.data.user
            return new UserModel(userData)
        })
        .catch(error => {
            console.log(error)
            return new UserModel(emptyUserData)
        })
}

function loadRepoData(username) {
    let query = `query {
                        user(login:"${username}") {
                            repositories(first: 100, orderBy: {field: UPDATED_AT, direction: DESC}){
                                nodes{
                                    name
                                    url
                                    description
                                    updatedAt
                                    primaryLanguage{
                                        name
                                        color
                                    }
                                    stargazers{
                                        totalCount
                                    }
                                    forkCount
                                }
                            }
                        }
                    }`
    githubQuery(query)
        .then(response => {
            let repoDataArray = response.data.user.repositories.nodes
            let repos = parseNodesToModel(repoDataArray, RepoModel)
            return repos
        })
        .catch(error => {
            console.log(error)
            return [new RepoModel(emptyRepoData)]
        });
}

function loadFollowerData(username) {
    let query = `
        {
          user(login: "${username}") {
            login
            bio
            followers(first: 100) {
              nodes {
                login
                bio
                avatarUrl
                repositories {
                  totalCount
                }
              }
            }
          }
        }`

    githubQuery(query)
        .then(response => {
            // console.log(response.data.user.followers.nodes)
            let userDataArray = response.data.user.followers.nodes
            let userList = parseNodesToModel(userDataArray, UserModel)
            userList.sort(function (a, b) {
                return a.publicReposCount - b.publicReposCount
            })
            return userList
        })
        .catch(error => {
            console.log(error)
            return [new UserModel(emptyUserData)]
        })
}

function loadFollowingData() {
    let query = `
        {
          user(login: "${this.currentUser}") {
            login
            bio
            following(first: 100) {
              nodes {
                login
                bio
                avatarUrl
                repositories {
                  totalCount
                }
              }
            }
          }
        }`

    githubQuery(query)
        .then(response => {
            // console.log(response.data.user.followers.nodes)
            let userDataArray = response.data.user.following.nodes
            let userList = parseNodesToModel(userDataArray, UserModel)
            userList.sort(function (a, b) {
                return a.publicReposCount - b.publicReposCount
            })
            return userList
        })
        .catch(error => {
            console.log(error)
            return [new UserModel(emptyUserData)]
        })
}

export {loadFollowerData, loadFollowingData, loadRepoData, loadUserData}