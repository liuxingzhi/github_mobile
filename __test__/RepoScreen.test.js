import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, {shallow, render, mount} from 'enzyme';
import toJson from "enzyme-to-json";
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({adapter: new Adapter()});

import {githubQuery} from '../utils/util';
import ProfileScreen from "../screens/Profile";
import RepositoryScreen from "../screens/Repo";
import FollowerScreen from "../screens/Following";
import FollowingScreen from "../screens/Follower";
import {RepoModel} from "../model/RepoModel";
import {UserModel} from "../model/UserModel";
import ShallowWrapper from "enzyme/imgs/ShallowWrapper";


const fakeRepoResponse = JSON.stringify(
    {
        "data": {
            "user": {
                "repositories": {
                    "nodes": [
                        {
                            "name": "ChromeDriverManager",
                            "url": "https://github.com/liuxingzhi/ChromeDriverManager",
                            "description": null,
                            "updatedAt": "2020-08-28T02:22:36Z",
                            "primaryLanguage": {
                                "name": "Python",
                                "color": "#3572A5"
                            },
                            "stargazers": {
                                "totalCount": 0
                            },
                            "forkCount": 0
                        },
                        {
                            "name": "course_register",
                            "url": "https://github.com/liuxingzhi/course_register",
                            "description": "crawl uiuc banner info",
                            "updatedAt": "2020-08-28T02:09:55Z",
                            "primaryLanguage": {
                                "name": "HTML",
                                "color": "#e34c26"
                            },
                            "stargazers": {
                                "totalCount": 1
                            },
                            "forkCount": 0
                        },
                        {
                            "name": "my_coursely",
                            "url": "https://github.com/liuxingzhi/my_coursely",
                            "description": null,
                            "updatedAt": "2020-08-28T01:51:41Z",
                            "primaryLanguage": {
                                "name": "Python",
                                "color": "#3572A5"
                            },
                            "stargazers": {
                                "totalCount": 1
                            },
                            "forkCount": 0
                        }
                    ]
                }
            }
        }
    }
)

// jest.setTimeout(60000)
test('test repo renders correctly', async () => {
    fetch.mockResponse(() => Promise.resolve(fakeRepoResponse))
    const wrapper =  shallow(<RepositoryScreen navigation={{addListener: () => null}}
                                              setCurrentUserHandler={(...args) => null}
                                              getCurrentUser={(...args) => null}
    />);
    expect(toJson(wrapper)).toMatchSnapshot();
})

test('test repo render when api is down', async () => {
    fetch.mockResponse(() => Promise.reject("API is down"))
    const wrapper = shallow(<RepositoryScreen navigation={{addListener: () => null}}
                                              setCurrentUserHandler={(...args) => null}
                                              getCurrentUser={(...args) => null}
    />);
    expect(toJson(wrapper)).toMatchSnapshot();
})

test('test repo render when api returns bad format data', async () => {
    fetch.mockResponse(() => Promise.resolve({}))
    const wrapper = shallow(<RepositoryScreen navigation={{addListener: () => null}}
                                              setCurrentUserHandler={(...args) => null}
                                              getCurrentUser={(...args) => null}
    />);
    expect(toJson(wrapper)).toMatchSnapshot();
})