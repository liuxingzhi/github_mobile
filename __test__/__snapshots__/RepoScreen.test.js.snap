// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test repo render when api is down 1`] = `
<Styled(Container)>
  <CustomerHeader
    navigation={
      Object {
        "addListener": [Function],
      }
    }
    setFilterCriteriaHandler={[Function]}
    title="Repositories"
  />
  <LoadingTheme
    visibility={true}
  />
  <Styled(Content)
    padder={true}
  >
    <Styled(Card)>
      <Styled(CardItem)
        bordered={true}
        header={true}
        key="header[object Object]"
        style={
          Object {
            "borderBottomWidth": 0,
            "flexWrap": "wrap",
          }
        }
      >
        <ForwardRef
          onPress={[Function]}
          style={
            Object {
              "flex": 3.2,
            }
          }
        >
          <Styled(Text) />
        </ForwardRef>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <Styled(Icon)
            name="staro"
            style={
              Object {
                "marginLeft": 30,
              }
            }
            type="AntDesign"
          />
          <Styled(Text)
            style={
              Object {
                "color": "#000000",
                "marginTop": 4,
              }
            }
          >
            0
          </Styled(Text)>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "marginLeft": 50,
              "marginRight": -10,
            }
          }
        >
          <Styled(Icon)
            name="code-fork"
            type="FontAwesome"
          />
          <Styled(Text)
            style={
              Object {
                "color": "#000000",
                "marginLeft": -7,
                "marginTop": 4,
              }
            }
          >
            0
          </Styled(Text)>
        </View>
      </Styled(CardItem)>
      <Styled(CardItem)
        key="body[object Object]"
      >
        <Styled(Text)>
          No description
        </Styled(Text)>
      </Styled(CardItem)>
      <Styled(CardItem)
        bordered={true}
        footer={true}
        key="footer[object Object]"
        style={
          Object {
            "borderTopWidth": 0,
          }
        }
      >
        <Styled(Left)>
          <Styled(Icon)
            name="primitive-dot"
            style={
              Object {
                "color": "#FFFFFF",
              }
            }
            type="Octicons"
          />
          <Styled(Text)>
            unknown
          </Styled(Text)>
        </Styled(Left)>
        <Styled(Right)>
          <Styled(Text)>
            last updated
          </Styled(Text)>
          <TimeAgo
            hideAgo={false}
            interval={60000}
            time={Date { NaN }}
          />
        </Styled(Right)>
      </Styled(CardItem)>
    </Styled(Card)>
  </Styled(Content)>
</Styled(Container)>
`;

exports[`test repo render when api returns bad format data 1`] = `
<Styled(Container)>
  <CustomerHeader
    navigation={
      Object {
        "addListener": [Function],
      }
    }
    setFilterCriteriaHandler={[Function]}
    title="Repositories"
  />
  <LoadingTheme
    visibility={true}
  />
  <Styled(Content)
    padder={true}
  >
    <Styled(Card)>
      <Styled(CardItem)
        bordered={true}
        header={true}
        key="header[object Object]"
        style={
          Object {
            "borderBottomWidth": 0,
            "flexWrap": "wrap",
          }
        }
      >
        <ForwardRef
          onPress={[Function]}
          style={
            Object {
              "flex": 3.2,
            }
          }
        >
          <Styled(Text) />
        </ForwardRef>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <Styled(Icon)
            name="staro"
            style={
              Object {
                "marginLeft": 30,
              }
            }
            type="AntDesign"
          />
          <Styled(Text)
            style={
              Object {
                "color": "#000000",
                "marginTop": 4,
              }
            }
          >
            0
          </Styled(Text)>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "marginLeft": 50,
              "marginRight": -10,
            }
          }
        >
          <Styled(Icon)
            name="code-fork"
            type="FontAwesome"
          />
          <Styled(Text)
            style={
              Object {
                "color": "#000000",
                "marginLeft": -7,
                "marginTop": 4,
              }
            }
          >
            0
          </Styled(Text)>
        </View>
      </Styled(CardItem)>
      <Styled(CardItem)
        key="body[object Object]"
      >
        <Styled(Text)>
          No description
        </Styled(Text)>
      </Styled(CardItem)>
      <Styled(CardItem)
        bordered={true}
        footer={true}
        key="footer[object Object]"
        style={
          Object {
            "borderTopWidth": 0,
          }
        }
      >
        <Styled(Left)>
          <Styled(Icon)
            name="primitive-dot"
            style={
              Object {
                "color": "#FFFFFF",
              }
            }
            type="Octicons"
          />
          <Styled(Text)>
            unknown
          </Styled(Text)>
        </Styled(Left)>
        <Styled(Right)>
          <Styled(Text)>
            last updated
          </Styled(Text)>
          <TimeAgo
            hideAgo={false}
            interval={60000}
            time={Date { NaN }}
          />
        </Styled(Right)>
      </Styled(CardItem)>
    </Styled(Card)>
  </Styled(Content)>
</Styled(Container)>
`;

exports[`test repo renders correctly 1`] = `
<Styled(Container)>
  <CustomerHeader
    navigation={
      Object {
        "addListener": [Function],
      }
    }
    setFilterCriteriaHandler={[Function]}
    title="Repositories"
  />
  <LoadingTheme
    visibility={true}
  />
  <Styled(Content)
    padder={true}
  >
    <Styled(Card)>
      <Styled(CardItem)
        bordered={true}
        header={true}
        key="header[object Object]"
        style={
          Object {
            "borderBottomWidth": 0,
            "flexWrap": "wrap",
          }
        }
      >
        <ForwardRef
          onPress={[Function]}
          style={
            Object {
              "flex": 3.2,
            }
          }
        >
          <Styled(Text) />
        </ForwardRef>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <Styled(Icon)
            name="staro"
            style={
              Object {
                "marginLeft": 30,
              }
            }
            type="AntDesign"
          />
          <Styled(Text)
            style={
              Object {
                "color": "#000000",
                "marginTop": 4,
              }
            }
          >
            0
          </Styled(Text)>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "marginLeft": 50,
              "marginRight": -10,
            }
          }
        >
          <Styled(Icon)
            name="code-fork"
            type="FontAwesome"
          />
          <Styled(Text)
            style={
              Object {
                "color": "#000000",
                "marginLeft": -7,
                "marginTop": 4,
              }
            }
          >
            0
          </Styled(Text)>
        </View>
      </Styled(CardItem)>
      <Styled(CardItem)
        key="body[object Object]"
      >
        <Styled(Text)>
          No description
        </Styled(Text)>
      </Styled(CardItem)>
      <Styled(CardItem)
        bordered={true}
        footer={true}
        key="footer[object Object]"
        style={
          Object {
            "borderTopWidth": 0,
          }
        }
      >
        <Styled(Left)>
          <Styled(Icon)
            name="primitive-dot"
            style={
              Object {
                "color": "#FFFFFF",
              }
            }
            type="Octicons"
          />
          <Styled(Text)>
            unknown
          </Styled(Text)>
        </Styled(Left)>
        <Styled(Right)>
          <Styled(Text)>
            last updated
          </Styled(Text)>
          <TimeAgo
            hideAgo={false}
            interval={60000}
            time={Date { NaN }}
          />
        </Styled(Right)>
      </Styled(CardItem)>
    </Styled(Card)>
  </Styled(Content)>
</Styled(Container)>
`;
